### PR TITLE
python311Packages.{jax,jaxlib,jaxlib-bin}: 0.4.23 -> 0.4.24

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://jax.readthedocs.io/en/latest/changelog.html#jax-0-4-24

cc @samuela 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 286948` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>12 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.aeppl</li>
    <li>python311Packages.aeppl.dist</li>
    <li>python311Packages.aesara</li>
    <li>python311Packages.aesara.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
  </ul>
</details>
<details>
  <summary>28 packages failed to build:</summary>
  <ul>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.awkward</li>
    <li>python311Packages.awkward.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.correctionlib</li>
    <li>python311Packages.correctionlib.dist</li>
    <li>python311Packages.dask-awkward</li>
    <li>python311Packages.dask-awkward.dist</li>
    <li>python311Packages.mplhep</li>
    <li>python311Packages.mplhep.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.torchrl</li>
    <li>python311Packages.torchrl.dist</li>
    <li>python311Packages.uproot</li>
    <li>python311Packages.uproot.dist</li>
    <li>python311Packages.vector</li>
    <li>python311Packages.vector.dist</li>
  </ul>
</details>
<details>
  <summary>17 packages built:</summary>
  <ul>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 286948` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>44 packages failed to build:</summary>
  <ul>
    <li>libretranslate (python311Packages.libretranslate)</li>
    <li>libretranslate.dist (python311Packages.libretranslate.dist)</li>
    <li>python311Packages.aeppl</li>
    <li>python311Packages.aeppl.dist</li>
    <li>python311Packages.aesara</li>
    <li>python311Packages.aesara.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.correctionlib</li>
    <li>python311Packages.correctionlib.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.vector</li>
    <li>python311Packages.vector.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>
<details>
  <summary>47 packages built:</summary>
  <ul>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.awkward</li>
    <li>python311Packages.awkward.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.dask-awkward</li>
    <li>python311Packages.dask-awkward.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib-bin</li>
    <li>python311Packages.jaxlib-bin.dist</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.mplhep</li>
    <li>python311Packages.mplhep.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.tensorflow-bin</li>
    <li>python311Packages.tensorflow-bin.dist</li>
    <li>python311Packages.torchrl</li>
    <li>python311Packages.torchrl.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.uproot</li>
    <li>python311Packages.uproot.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/jax/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/jax/default.nix:1]: Already broken on hydra: arviz, pymc, aesara, probably more-when are we going to extend nixpkgs-review to check hydra before posting

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jax/default.nix:1]: @GaetanLepage this doesn't look right: https://github.com/NixOS/nixpkgs/pull/286948/checks?check_run_id=21315293672

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jax/default.nix:1]: Not a regression compared to master:
- aarch64-darwin has been failing to evaluate: https://hydra.nixos.org/job/nixpkgs/trunk/python311Packages.jaxlib.aarch64-darwin
- x86_64-darwin fails with a dependency: https://hydra.nixos.org/job/nixpkgs/trunk/python311Packages.jaxlib.x86_64-darwin

Comment (SomeoneSerge) [reply pkgs/development/python-modules/jax/default.nix:1]: Is this where the "ofborg-internal-error" label comes from?

File: pkgs/development/python-modules/jaxlib/default.nix [hunk 1]
```diff
@@ -195,7 +195,11 @@ let
     '';
 
     bazelRunTarget = "//jaxlib/tools:build_wheel";
-    runTargetFlags = [ "--output_path=$out" "--cpu=${arch}" ];
+    runTargetFlags = [
+      "--output_path=$out"
+      "--cpu=${arch}"
+      "--jaxlib_git_hash=${src.outputHash}"
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/jaxlib/default.nix:201]: Yes, it seems to work fine.

File: pkgs/development/python-modules/jax/default.nix [hunk 2]
```diff
@@ -89,6 +89,8 @@ buildPythonPackage rec {
     "testKde3"
     "testKde5"
     "testKde6"
+    # Invokes python manually in a subprocess, which does not have the correct dependencies
```
Comment (samuela) [inline pkgs/development/python-modules/jax/default.nix:92]: ooc what are the missing dependencies?

File: pkgs/development/python-modules/jaxlib/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (samuela) [reply pkgs/development/python-modules/jaxlib/default.nix:201]: iiuc that's not a nix argument. it ends up mapping here:
https://github.com/google/jax/blob/4c505f8bac45517970149b13343988090fdaa920/jaxlib/tools/build_wheel.py#L45-L50

with that in mind we should instead do
```suggestion
      "--jaxlib_git_hash=${src.rev}"
```

File: pkgs/development/python-modules/jax/default.nix (hunk 2 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/jax/default.nix:92]: It fails at importing `jaxlib.version` if I remember well.
Basically, I was able to succesfully execute the test artificially in `preCheck`. But the exact same snippet was failing in `pytest` when using `subprocess`.

Comment (GaetanLepage) [reply pkgs/development/python-modules/jax/default.nix:92]: However, I checked (by `print`ing) that [this python](https://github.com/google/jax/blob/jaxlib-v0.4.24/tests/logging_test.py#L72) was exactly the same as the one I was using in the `preCheck`.

Comment (samuela) [reply pkgs/development/python-modules/jax/default.nix:92]: must be something weird with pytest and subprocess then?

**Comment by GaetanLepage** [comment]
@samuela I switched to using `src.rev` for `--jaxlib_git_hash`.

**Comment by samuela** [comment]
We may need to rebase after https://github.com/NixOS/nixpkgs/pull/285037

**Comment by GaetanLepage** [comment]
> We should prob rebase after #285037

Done :)

File: pkgs/development/python-modules/jaxlib/bin.nix [hunk 3]
```diff
@@ -130,19 +130,19 @@ let
...
-      hash = "sha256-dMUcRnHjl8NyUeO3P1x7CNgF0iAHFKIzUtHh+/CNkow=";
+      hash = "sha256-Ipy3vk1yUplpNzECAFt63aOIhgEWgXG7hkoeTIk9bQQ=";
     };
     "cuda12.2-3.12" = fetchurl {
       url = "https://storage.googleapis.com/jax-releases/cuda12/jaxlib-${version}+cuda12.cudnn89-cp312-cp312-manylinux2014_x86_64.whl";
-      hash = "sha256-kXJ6bUwX+QybqYPV9Kpwv+lhdoGEFRr4+1T0vfXoWRo=";
+      hash = "sha256-LSnZHaUga/8Z65iKXWBnZDk4yUpNykFTu3vukCchO6Q=";
     };
     "cuda11.8-3.9" = fetchurl {
       url = "https://storage.googleapis.com/jax-releases/cuda11/jaxlib-${version}+cuda11.cudnn86-cp39-cp39-manylinux2014_x86_64.whl";
```
Comment (samuela) [inline pkgs/development/python-modules/jaxlib/bin.nix:148]: we'll also need to update the hashes for the `cuda11.8-*` series of keys

**Review by samuela — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/jaxlib/bin.nix (hunk 3 above)
Comment (GaetanLepage) [reply pkgs/development/python-modules/jaxlib/bin.nix:148]: Done :)

File: pkgs/development/python-modules/jaxlib/default.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply pkgs/development/python-modules/jaxlib/default.nix]: Done ! Still works.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 286948` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>44 packages failed to build:</summary>
  <ul>
    <li>libretranslate (python311Packages.libretranslate)</li>
    <li>libretranslate.dist (python311Packages.libretranslate.dist)</li>
    <li>python311Packages.aeppl</li>
    <li>python311Packages.aeppl.dist</li>
    <li>python311Packages.aesara</li>
    <li>python311Packages.aesara.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.correctionlib</li>
    <li>python311Packages.correctionlib.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.vector</li>
    <li>python311Packages.vector.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>
<details>
  <summary>47 packages built:</summary>
  <ul>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.awkward</li>
    <li>python311Packages.awkward.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.dask-awkward</li>
    <li>python311Packages.dask-awkward.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib-bin</li>
    <li>python311Packages.jaxlib-bin.dist</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.mplhep</li>
    <li>python311Packages.mplhep.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
    <li>python311Packages.tensorflow-bin</li>
    <li>python311Packages.tensorflow-bin.dist</li>
    <li>python311Packages.torchrl</li>
    <li>python311Packages.torchrl.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.uproot</li>
    <li>python311Packages.uproot.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 286948` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages marked as broken and skipped:</summary>
  <ul>
    <li>python311Packages.dm-sonnet</li>
    <li>python311Packages.dm-sonnet.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.tensorflow-datasets</li>
    <li>python311Packages.tensorflow-datasets.dist</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>38 packages failed to build:</summary>
  <ul>
    <li>libretranslate (python311Packages.libretranslate)</li>
    <li>libretranslate.dist (python311Packages.libretranslate.dist)</li>
    <li>python311Packages.aeppl</li>
    <li>python311Packages.aeppl.dist</li>
    <li>python311Packages.aesara</li>
    <li>python311Packages.aesara.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.bambi</li>
    <li>python311Packages.bambi.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.correctionlib</li>
    <li>python311Packages.correctionlib.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.pymc</li>
    <li>python311Packages.pymc.dist</li>
    <li>python311Packages.pytensor</li>
    <li>python311Packages.pytensor.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.vector</li>
    <li>python311Packages.vector.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>53 packages built:</summary>
  <ul>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.arviz</li>
    <li>python311Packages.arviz.dist</li>
    <li>python311Packages.awkward</li>
    <li>python311Packages.awkward.dist</li>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.dask-awkward</li>
    <li>python311Packages.dask-awkward.dist</li>
    <li>python311Packages.equinox</li>
    <li>python311Packages.equinox.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.gymnasium</li>
    <li>python311Packages.gymnasium.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib-bin</li>
    <li>python311Packages.jaxlib-bin.dist</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.mplhep</li>
    <li>python311Packages.mplhep.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.tensorflow-bin</li>
    <li>python311Packages.tensorflow-bin.dist</li>
    <li>python311Packages.torchrl</li>
    <li>python311Packages.torchrl.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.uproot</li>
    <li>python311Packages.uproot.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>

**Review by SomeoneSerge — APPROVED** [review]

File: pkgs/development/python-modules/jax/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/jax/default.nix:1]: > dependencies of numpyro, https://logs.ofborg.org/?key=nixos/nixpkgs.286948&attempt_id=fdcc94b9-0437-47b0-ab64-0c808fcdc01b

I don't understand this interface at all, but I suspect it's pytorch from master failing: https://hydra.nixos.org/build/248981792

Comment (SomeoneSerge) [inline pkgs/development/python-modules/jax/default.nix:1]: @samuela your request for changes is pinned; I'd like to have this merged, any objections?

File: pkgs/development/python-modules/jaxlib/default.nix [hunk 4]
```diff
@@ -195,7 +195,12 @@ let
     '';
 
     bazelRunTarget = "//jaxlib/tools:build_wheel";
-    runTargetFlags = [ "--output_path=$out" "--cpu=${arch}" ];
+    runTargetFlags = [
+      "--output_path=$out"
+      "--cpu=${arch}"
+      # This has no impact whatsoever...
+      "--jaxlib_git_hash='12345678'"
```
Comment (samuela) [inline pkgs/development/python-modules/jaxlib/default.nix:202]: can we just skip it then? i'd prefer to not insert magical numbers if we can avoid

or why not use `src.rev`?

File: pkgs/development/python-modules/jax/default.nix (hunk 2 above)
Comment (samuela) [reply pkgs/development/python-modules/jax/default.nix:92]: nit: can we make this comment more specific? "does not have the correct dependencies" indicates to me that we should either identify the correct dependencies and use them, or specify what dependencies we are missing and why they can't be added

**Review by samuela — COMMENTED** [review]
just 2 small things

File: pkgs/development/python-modules/jax/default.nix
Comment (samuela) [reply pkgs/development/python-modules/jax/default.nix:1]: reviewed!

File: pkgs/development/python-modules/jax/default.nix (hunk 2 above)
Comment (SomeoneSerge) [reply pkgs/development/python-modules/jax/default.nix:92]: Um sounds like upstream messing with PYTHONPATH and subprocesses

File: pkgs/development/python-modules/jaxlib/default.nix (hunk 4 above)
Comment (SomeoneSerge) [reply pkgs/development/python-modules/jaxlib/default.nix:202]: Alternatives considered: `0000000000000000000000000000000000000000` 😆

File: pkgs/development/python-modules/jax/default.nix (hunk 2 above)
Comment (SomeoneSerge) [inline pkgs/development/python-modules/jax/default.nix:92]: @samuela I don't think figuring out upstream's subprocess/pythonpath pollution issues is a necessity for us. Shall we let this through?

Comment (samuela) [reply pkgs/development/python-modules/jax/default.nix:92]: ok fair enough, fine with me

**Review by samuela — APPROVED** [review]
unblocking merge

**Comment by samuela** [comment]
merging since i was the last remaining blocker and it builds and runs fine for me

**Pull request merged**